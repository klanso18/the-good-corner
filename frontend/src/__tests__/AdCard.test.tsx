import "@testing-library/jest-dom";
import { render, screen } from "@testing-library/react";
import AdCard from "../components/AdCard";

describe("AdCard component", () => {
  it("renders the AdCard component", () => {
    const view = render(
      <AdCard
        ad={{
          id: 1,
          title: "hello",
          price: 10,
          picture:
            "https://m.media-amazon.com/images/I/51TJ3ERPBwL._AC_UF1000,1000_QL80_.jpg",
        }}
        link="/"
      />
    );
    const title = screen.getByText("hello");
    expect(title).toBeInTheDocument();
    expect(screen.getByText(/10/)).toBeInTheDocument();
    expect(view.baseElement).toMatchSnapshot();
  });
});
