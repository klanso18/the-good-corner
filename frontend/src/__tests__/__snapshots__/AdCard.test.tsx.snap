// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AdCard component renders the AdCard component 1`] = `
<body>
  <div>
    <div
      class="w-[400px]"
    >
      <a
        href="/"
      >
        <div
          class="shadow-md border rounded-lg  p-6 bg-white mr-3 mb-3"
        >
          <img
            alt="ad picture"
            class="h-[200px] w-full object-cover rounded-md"
            src="https://m.media-amazon.com/images/I/51TJ3ERPBwL._AC_UF1000,1000_QL80_.jpg"
          />
          <div
            class="flex justify-between pt-6"
          >
            <div
              class="ad-card-title"
            >
              hello
            </div>
            <div
              class="ad-card-price"
            >
              10
               â‚¬
            </div>
          </div>
        </div>
      </a>
    </div>
  </div>
</body>
`;
